File: Queue/QueueArray/lib/queue.c
#include <stdio.h>
#include <stdlib.h>
#include "queue.h"

//Inicializa a fila
void initializeQueue(QUEUE* pq){
    pq->front = -1;
    pq->rear = -1;
    pq->size = 0;
}

//Incere no final da fila
void enqueue(QUEUE* pq, ITEM item){
    if(pq->size == SIZE){
        printf("Queue is full!!\n");
        exit(-1);
    }

    if(pq->rear == SIZE - 1){
        pq->rear = -1;
    }

    ++(pq->size);
    pq->item[++(pq->rear)] = item;
}

//Retira do começo da fila
ITEM dequeue(QUEUE* pq){
    if(empty(pq)){
        printf("Queue is empty!!\n");
        exit(-1);
    }

    if(pq->front == SIZE - 1){
        pq->front = -1;
    }

    ITEM temp = pq->item[++(pq->front)];

    --(pq->size);
    return temp;
}

//Verifica se a fila está vazia
int empty(QUEUE* pq){
    if(pq->size == 0){
        return TRUE;
    }else{
        return FALSE;
    }
}

File: Lists/Queue/lib/Queue.java
//Pacote em que a classe está
package lib;

//Classe Queue genérica que herada da lista encadeada também generica
public class Queue<T extends Comparable<T>> extends List<T>{
    //Retorna true se a fila está vazia
    public boolean isEmpty(){
        if(getSizeList() == 0){
            return true;
        }else{
            return false;
        }
    }//Fim isEmpty

    //Retorna o primero da fila
    public T qFront(){
        T temp = removeFront();

        setFrontList(temp);

        return temp;
    }//Fim qFront

    //Mostra o tamanho da fila
    public int qSize(){
        return getSizeList();
    }
    //Fim qSize

    //Metodo enqueue que coloca um objeto no final da fila
    public void enqueue(T data){
        setRearList(data);
    }//Fim enqueue

    //Metodo dequeue que remove um objeto do começo da fila
    public T dequeue(){
        if(getSizeList() == 0){
            System.out.println("Error in dequeue method, queue is empty!!\n\n");
            System.exit(-1);
        }

        return removeFront();
    }//Fim dequeue
}//Fim classe Queue